src/main/java/org/gingesnap/k8s/CacheRuleWatcher.java

```diff
@@ -0,0 +1,61 @@
+package org.gingesnap.k8s;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.Watch;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.WatcherException;
+import io.quarkus.arc.lookup.LookupIfProperty;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+@Singleton
+public class CacheRuleWatcher {
+
+   private static final Logger log = LoggerFactory.getLogger(CacheRuleWatcher.class);
+   private static final String EAGER_CACHE_RULE_NAME = "EagerCacheRule";
+
+   @Inject Instance<KubernetesClient> client;
```

Comment:
Why do we need to use `Instance<>` here?

We should just be able todo.
```
   @Inject
   KubernetesClient client;
```

Reply:
That's because of the `@LookupIfProperty` in `KubernetesClientProducer`. Seems like that annotation only works when wrapped around an `Instance`.

Reply:
Only because we're creating a way for the application to start without needing to configure the stuff related to kubernetes.

Reply:
Gotcha, I wasn't aware of that property :+1: